src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/RegistrationValidators.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    public class RegistrationValidators
+    {
+        public static string GetContainsString(bool contains)
+        {
+            return contains ? "contains" : "doesn't contain";
+        }
+        
+        /// <summary>
+        /// Verifies that the index.json blob for the package exists when it should.
+        /// </summary>
+        public class RegistrationIndexExistsValidator : Validator, IValidator<RegistrationEndpoint>
+        {
+            public RegistrationIndexExistsValidator(
+                IDictionary<FeedType, SourceRepository> feedToSource,
+                ILogger<RegistrationIndexExistsValidator> logger)
+                : base(feedToSource, logger)
+            {
+                v2Metadata = feedToSource[FeedType.HttpV2].GetResource<MetadataResource>();
+                v3Metadata = feedToSource[FeedType.HttpV3].GetResource<MetadataResource>();
+
+                regResource = feedToSource[FeedType.HttpV3].GetResource<RegistrationResourceV3>();
+            }
+
+            private readonly MetadataResource v2Metadata;
+            private readonly MetadataResource v3Metadata;
+
+            private readonly RegistrationResourceV3 regResource;
+
+            protected override async Task RunInternal(ValidationContext data)
+            {
+                var v2PackageIdExists = await v2Metadata.Exists(data.Package.Id, includePrerelease: true, includeUnlisted: true, log: Logger.AsCommon(), token: data.CancellationToken);
```

Comment:
>AsCommon [](start = 141, length = 8)

No point in creating the wrapper over and over right? Couldn't there be Logger and CommonLogger properties?